File: app/src/main/java/com/hypnoweb/hypnowebapp/splashscreen/SplashScreenFragment.java
package com.hypnoweb.hypnowebapp.splashscreen;


import android.arch.lifecycle.ViewModelProvider;
import android.arch.lifecycle.ViewModelProviders;
import android.databinding.DataBindingUtil;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.hypnoweb.hypnowebapp.R;
import com.hypnoweb.hypnowebapp.databinding.SplashscreenfragmentBinding;
import com.hypnoweb.hypnowebapp.di.interfaces.Injectable;

import javax.inject.Inject;


public class SplashScreenFragment extends Fragment implements Injectable {

    @Inject
    ViewModelProvider.Factory viewModelFactory;


    public static SplashScreenFragment getInstance() {
        return new SplashScreenFragment();
    }

    public SplashScreenFragment() {
    }

    @Nullable
    @Override
    public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        return inflater.inflate(R.layout.splashscreenfragment, container, false);
    }

    @Override
    public void onActivityCreated(@Nullable Bundle savedInstanceState) {
        super.onActivityCreated(savedInstanceState);
        final SplashViewModel viewModel = ViewModelProviders.of(getActivity(), viewModelFactory)
                .get(SplashViewModel.class);
        SplashscreenfragmentBinding binding = DataBindingUtil.bind(getView());
        binding.setViewModel(viewModel);

    }

}

File: app/src/main/java/com/hypnoweb/hypnowebapp/splashscreen/SplashScreenFragment.kt
package com.hypnoweb.hypnowebapp.splashscreen

import android.arch.lifecycle.ViewModelProviders
import android.databinding.DataBindingUtil
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import com.hypnoweb.hypnowebapp.R
import com.hypnoweb.hypnowebapp.databinding.SplashscreenfragmentBinding
import com.hypnoweb.hypnowebapp.di.ViewModelFactory.ViewModelFactory
import dagger.android.support.DaggerFragment
import javax.inject.Inject

class SplashScreenFragment : DaggerFragment() {

    @Inject
    lateinit var viewModelFactory: ViewModelFactory

    override fun onCreateView(inflater: LayoutInflater?, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        return inflater!!.inflate(R.layout.splashscreenfragment, container, false)
    }

    override fun onActivityCreated(savedInstanceState: Bundle?) {
        super.onActivityCreated(savedInstanceState)
        val viewModel = ViewModelProviders.of(activity, viewModelFactory)
                .get(SplashViewModel::class.java)
        val binding = DataBindingUtil.bind<SplashscreenfragmentBinding>(view!!)
        binding!!.viewModel = viewModel

    }

    companion object {
        fun getInstance(): SplashScreenFragment {
            val fragment = SplashScreenFragment()
            return fragment

        }
    }

}
